tests/test_core/test_core_provision.py

```diff
@@ -0,0 +1,280 @@
+"""Test suite to test the core matcha provision function."""
```

Comment:
Some additional unhappy path tests could be added in here. For example, a test for when the remote state is stale and checking that expected behaviour happens - see above comment about possible changes to that part of the provisioning.